### ISSUE TITLE
Make Node type param optional everywhere

### ISSUE THREAD

**Issue by japgolly** [japgolly/scalajs-react#230]
For example, if I'm wrapping a JS component I need to type:

``` scala
JsComponentType[P, S, TopNode]
```

The `Node` param is only occasionally used and is just noise everywhere. Instead something this would be better:

``` scala
// Can specify without
JsComponentType[P, S]

// Can specify if desired.
JsComponentTypeWithNode[P, S, Node]
```

and of course, make Scalac understand the relationship between the two so user never has to distinguish or translate.

**Comment by japgolly** [comment]
This will be fixed as part of #259.

**Comment by japgolly** [comment]
Ditching the `N` type. It's not worth it.
